Adds Tests for coverage, supporting for working with targets and a fix

Sqeaky opened the pull request:
for a bug that was found.

This might feel like a lot of code, but WorkArounds.rb was simply extracted from RootTest.rb, The 3 CMakeLists.txt differ by 3 lines apiece, Jagati.cmake gained a single if and some tabbing.

The most dense real code changes are in CMakeTargets.rb. It loads a the names of targets from a CMake created file and stores them in an array. Ruby arrays have significant tools for inspecting their contents so I rigged the CMakeTargets class to pass most method calls of onto the array. The route of using method_missing is generally considered poor for performance because it interferes with the method cache, but that also doesn't matter until you are servicing thousands of requests per second so I figured it was a reasonable expedient in these tests.

Let me know if you have any questions.

Sqeaky committed:
Adds Tests for coverage, supporting for working with targets and a fix
for a bug that was found.

BTSJenkins commented:
This passes all the automated tests I think this is good - ☺

You humans can take a look at at the build here: http://blacktopp.ddns.net:8080/job/Jagati/45/

Sqeaky commented:
This will close issue #30 and #35 if it passes.

Sqeaky committed:
Troubleshooting changes to look at Broken Apple in Travis

BTSJenkins commented:
This passes all the automated tests I think this is good - ☺

You humans can take a look at at the build here: http://blacktopp.ddns.net:8080/job/Jagati/46/

Sqeaky committed:
An attempt to fix Apple/Clang issues by correctly reporting Clang coverage

BTSJenkins commented:
This passes all the automated tests I think this is good - ☺

You humans can take a look at at the build here: http://blacktopp.ddns.net:8080/job/Jagati/47/

Sqeaky committed:
Removed debugging output and Found/fixed bug with Clang coverage detection.

Per clang docs: https://clang.llvm.org/docs/SourceBasedCodeCoverage.html

--coverage which is what we use with GCC and clang supports it as well.

BTSJenkins commented:
This passes all the automated tests I think this is good - ☺

You humans can take a look at at the build here: http://blacktopp.ddns.net:8080/job/Jagati/48/

Sqeaky commented:
I made issue #36 to describe the bug I found and fixed with this last round of tests.
